### PR TITLE
Use fakeclock correctly in dockershim unit test.

### PR DESCRIPTION
**Pull request by Random-Liu**
Fixes https://github.com/kubernetes/kubernetes/issues/33428.

/cc @feiskyer @yujuhong

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33448)

### CODE REVIEW

**Comment by feiskyer** [comment]
@k8s-bot gci gke e2e test this

**Comment by feiskyer** [comment]
@k8s-bot e2e test this

File: pkg/kubelet/dockershim/docker_service_test.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ import (
 )
 
 func newTestDockerSevice() (*dockerService, *dockertools.FakeDockerClient, *clock.FakeClock) {
-	c := dockertools.NewFakeDockerClient()
 	fakeClock := clock.NewFakeClock(time.Time{})
+	c := dockertools.NewFakeDockerClientWithClock(fakeClock)
```
Comment (yujuhong) [inline pkg/kubelet/dockershim/docker_service_test.go]: uh..forgot to change this. Thanks

File: pkg/kubelet/dockershim/docker_container_test.go [hunk 2]
```diff
@@ -123,7 +123,7 @@ func TestContainerStatus(t *testing.T) {
 	}
 
 	// Create the container.
-	fClock.SetTime(time.Now())
+	fClock.SetTime(time.Now().Add(-1 * time.Hour))
```
Comment (yujuhong) [inline pkg/kubelet/dockershim/docker_container_test.go]: why -1 hour?

**Comment by yujuhong** [comment]
@k8s-bot gke e2e test this issue: #IGNORE

File: pkg/kubelet/dockershim/docker_container_test.go (hunk 2 above)
Comment (Random-Liu) [inline pkg/kubelet/dockershim/docker_container_test.go]: I just want to make sure `Create`, `Start` and `Finish` timestamps are different.
It seems that they may be the same, that's why the test could pass before.

Comment (yujuhong) [inline pkg/kubelet/dockershim/docker_container_test.go]: sounds fair

**Comment by yujuhong** [comment]
Marking p1 for fixing flakes.

**Pull request merged**